pallets/messages/src/migration/v2.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{BlockNumberFor, Config, Message, MessagesV2, Pallet, SchemaId, LOG_TARGET};
+use frame_support::{pallet_prelude::*, storage_alias, traits::OnRuntimeUpgrade, weights::Weight};
+use log;
+use sp_runtime::Saturating;
+
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+/// old structures and storages
+pub mod old {
+	use super::*;
+	use common_primitives::msa::MessageSourceId;
+	use sp_std::fmt::Debug;
+
+	/// old message structure that was stored
+	#[derive(Default, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+	#[scale_info(skip_type_params(MaxDataSize))]
+	#[codec(mel_bound(MaxDataSize: MaxEncodedLen))]
+	pub struct OldMessage<MaxDataSize>
+	where
+		MaxDataSize: Get<u32> + Debug,
+	{
+		///  Data structured by the associated schema's model.
+		pub payload: BoundedVec<u8, MaxDataSize>,
+		/// Message source account id of the Provider. This may be the same id as contained in `msa_id`,
+		/// indicating that the original source MSA is acting as its own provider. An id differing from that
+		/// of `msa_id` indicates that `provider_msa_id` was delegated by `msa_id` to send this message on
+		/// its behalf.
+		pub provider_msa_id: MessageSourceId,
+		///  Message source account id (the original source).
+		pub msa_id: Option<MessageSourceId>,
+		///  Stores index of message in block to keep total order.
+		pub index: u16,
+	}
+
+	/// old permanent storage for messages mapped by block number and schema id.
+	#[storage_alias]
+	pub(crate) type Messages<T: Config> = StorageDoubleMap<
```

Comment:
How does storage alias know what prefix to use for the old storage? Going back over this I expected something like `#[crate::storage_alias(pallet_name)]`

Reply:
It gets it from the name of the Storage in this case `Messages<T: Config>`. This name should stay the same

Reply:
I tried it with try-run-time and it worked, will add a test to double verify.

Reply:
I think it will look for the same prefix since it's still inside the crate.

Reply:
Added a test and worked as expected. As mentioned the `storage-name `comes from the name of type and the `pallet-name` comes from the generic params Pallet<T>